### ISSUE TITLE
x/tools/cmd/goyacc: obtain value of the start expression without global variables

### ISSUE THREAD

**Issue by amlinux** [golang/go#20861]
### What did you do?

My use case: parse an expression and return the syntax tree. Here is a fragment of the grammar:

```
%start expr
%%
expr: NUM { $$ = numToExpr($1) }
%%
```

It's invoked as:
```
if exprParse(&exprLex{...}) != 0  {
  return errors.New(...)
}
// ??? Here I'd like to have an ability to extract the top level value.
```

### What did you expect to see?

I would expect some goroutine-safe way how to extract value of the top-level expression ("expr" in the example above).

### What did you see instead?

In the C world the recommended approach is to introduce a global variable and have a top-level rule with only one purpose - to write to the global variable. In Go it looks like this:

```
%start top
%%
top: expr { result = $1 }  # It's a global variable!
expr: NUM { $$ = numToExpr($1) }
%%
```

It's not goroutine-safe. goyacc should provide some interface to extract the top level value.

**Comment by cznic** [comment]
There's no absolutely need to use a static variable for this, use the yylex local variable.

**Comment by amlinux** [comment]
Indeed. yylex.(*yyLex) allows to get the syntax outside, thanks for the idea!
Don't you think passing data through lexer is hacky workaround?
